dev/CMakeLists.txt

```diff
@@ -109,3 +109,15 @@ target_link_libraries(${EXE_NAME}
   ${SDL_LIBRARIES}
   ${ADDITIONAL_LIBRARIES}
   ${OPENGL_LIBRARIES})
+
+if(UNIX)
+   include(GNUInstallDirs)
+   set(BINDIR ${CMAKE_INSTALL_FULL_BINDIR})
+   set(DATADIR ${CMAKE_INSTALL_FULL_DATADIR})
+else()
```

Comment:
remove `else` case? We don't want to add the current dir as `DATADIR`.

Reply:
I'm thinking of Windows here, so you can do something like this (untested)

```
cmake -DCMAKE_INSTALL_PREFIX=distribution .
make install
```

And end up with a `distribution` directory containing the required bits.

Reply:
On non-unix platforms, we're currently not using CMake, and we'd like to keep the directory structure as-is. E.g. both VS and Xcode produce an executable in the `lobster` dir. They don't need this additional data dir.

Reply:
Should I just remove the whole `if` and unconditionally use `GNUInstallDirs`?

Reply:
No, again, don't want this to affect other platforms.

Reply:
Which non-unix platforms are built via CMake and what should a `make install` look like on those?

Reply:
None.
To be clear, we have these 6 platforms:

Windows: VS
OS X & iOS: XCode
Android: Android SDK/NDK
WebAssembly: Makefile
Linux: CMake

Theoretically other unices might build with the CMake file, but I don't think anyone has ever tried.

I may want to switch Windows to CMake at some point, but when I do, it shouldn't be trying to install files anywhere, just use the files in-place. So make install in system dir code only take effect on Linux for now.

Reply:
What about this solution? I haven't tested, but on Unix installing to the sources dir seems to work. Files won't be copied (src and dst are the same), so only the lobster executable will be copied on `make install`.

Reply:
Again, on platforms that are non a Linux-alike (which is most of them), the existing behavior needs to be preserved. This changes the Windows exe for example to not be in `bin` anymore, but in some install dir? There is no "installing" on this platform. Please make your changes affect Linux only. I thought the whole point of this PR was just to allow us to do Linux packaging? So lets not use `UNIX` as it also includes all Apple platforms.

Reply:
- You said only Linux uses CMake.
- Even if you switched the Windows build to CMake, this behaviour is more reasonable. It should generate the lobster binary in the checkout directory. If you no longer want that behaviour, it should be easy to put it in a `bin` directory. 
- How can an `install` target for non-Unix be a bad thing? It should allow an easy isolation of required files for distribution (everything except the `dev` folder). If you just want to work in an install-less setup, just don't do `make install` and there you are.

Reply:
It currently does.. even so, OS-specific stuff should be in the proper conditionals. Even on Linux, it should end up in the `bin` dir during development, when not explicitly using `make install`. And there's nothing "more reasonable" given such a wide diversity of OS standards. The concept of "install" in this sense doesn't make sense for Windows (unless it generated an installer).

Reply:
It's already generating the unix executables under `bin`, check the last commit. And I have to disagree, `install` makes sense on Windows. Just tell it where to generate a distributable folder with `-DCMAKE_INSTALL_PREFIX=distributable`.

Reply:
Ok, I guess we disagree. If you insist, it can go in, I can always change it once I get to use it on Windows.